app/models/standard_identifiers.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+
+class StandardIdentifiers
+  attr_reader :identifiers
+
+  def initialize(term)
+    @identifiers = {}
+    term_pattern_checker(term)
+    strip_invalid_issns
+  end
+
+  private
+
+  def term_pattern_checker(term)
+    term_patterns.each_pair do |type, pattern|
+      @identifiers[type.to_sym] = match(pattern, term) if match(pattern, term).present?
+    end
+  end
+
+  # Note on the limitations of this implementation
+  # We only detect the first match of each pattern, so a search of "1234-5678 5678-1234" will not return two ISSNs as
+  # might be expected, but just "1234-5678". Using ruby's string.scan(pattern) may be worthwhile if we want to detect
+  # all possible matches instead of just the first. That may require a larger refactor though as initial tests of doing
+  # that change did result in unintended results so it was backed out for now.
+  def match(pattern, term)
+    pattern.match(term).to_s.strip
+  end
+
+  # term_patterns are regex patterns to be applied to the basic search box input
+  def term_patterns
```

Comment:
Would it be useful to detect handles? (To be clear, I'm not requesting a change; just curious.)

Reply:
Interesting idea. I don't know enough about handles to know if this would be useful useful, or just theoretically useful. Before resolving this conversation we can open a ticket to explore this further.

Reply:
Good idea. I haven't really done any analysis of our search logs, so I think a spike is in order: https://mitlibraries.atlassian.net/browse/ENGX-245